scripts/test-bump-version.sh

```
@@ -0,0 +1,163 @@
+
+assert_fail() {
+    echo -e "${RED}✗ FAIL: $1${NC}"
+    ((TESTS_FAILED++))
+}
+
+setup_test_dir() {
+    local test_num="$1"
+    local test_dir="/tmp/bump_version_test_$test_num"
+    rm -rf "$test_dir"
+    mkdir -p "$test_dir"
+    cd "$test_dir"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)